lib/teams/aws-batch/aws-batch-on-eks-team.ts

```diff
@@ -0,0 +1,167 @@
+
+    const AwsBatchAddOn = clusterInfo.getProvisionedAddOn('AwsBatchAddOn');
+
+    if (AwsBatchAddOn === undefined) {
+      throw new Error("AwsBatchAddOn must be deployed before creating AWS Batch on EKS team.");
+    }
+
+    // Set AWS Batch namespace and necessary RBACs
+    const ns = this.setBatchEksNamespace(clusterInfo, this.batchTeam.namespace);
+
+    // Create compute environment
+    const computeEnv = this.setComputeEnvironment(clusterInfo, this.batchTeam.namespace, ns, allocStr);
+
+    // Submit a job queue
+    const jobQueue = new batch.CfnJobQueue(clusterInfo.cluster.stack,'batch-eks-job-queue',{
+      jobQueueName: this.batchTeam.jobQueueName,
+      priority: this.batchTeam.priority,
+      computeEnvironmentOrder: [
+        {
+          order: 1,
+          computeEnvironment: computeEnv.attrComputeEnvironmentArn
+        }
+      ]
+    })
+
+    jobQueue.node.addDependency(computeEnv);
+  }
+  /**
+   * method to to apply k8s RBAC to the service account used by Batch service role
+   * @param ClusterInfo EKS cluster where to apply the RBAC
+   * @param namespace Namespace where the RBAC are applied
+   * @param createNamespace flag to create namespace if not already created
+   * @returns 
+   */
+
+  private setBatchEksNamespace(clusterInfo: ClusterInfo, namespace: string): eks.KubernetesManifest {
+
+    let doc = readYamlDocument(`${__dirname}/aws-batch-rbac-config.ytpl`);
+
+    //Get the RBAC definition and replace with the namespace provided by the user
+    const manifest = doc.split("---").map(e => loadYaml(e));
+
+    const values: Values = {
+      namespace: namespace
+    };
+
+    const manifestDeployment: ManifestDeployment = {
+      name: 'aws-batch-rbacs',
+      namespace: namespace,
+      manifest,
+      values
+    };
+
+    const kubectlProvider = new KubectlProvider(clusterInfo);
+    const statement = kubectlProvider.addManifest(manifestDeployment);
+
+    const batchEksNamespace = createNamespace(namespace, clusterInfo.cluster, true, true);
+    statement.node.addDependency(batchEksNamespace);
+
+    return batchEksNamespace;
+  }
+
+  private setComputeEnvironment(clusterInfo: ClusterInfo, namespace: string, nsNode: eks.KubernetesManifest, allocStr: string): batch.CfnComputeEnvironment {
+    const nodeGroups = assertEC2NodeGroup(clusterInfo, "Batch Compute Environment");
+    const ngRoleNames = nodeGroups.map((ng: eks.Nodegroup | AutoScalingGroup) => {return ng.role.roleName});
+    const cluster = clusterInfo.cluster;
+    const ngRole = ngRoleNames[0];
+
+    // Need to create instance profile for the nodegroup role
+    const instanceProfile = new iam.CfnInstanceProfile(cluster, 'ng-role-instance-profile',{
+      instanceProfileName: ngRole,
+      roles: [ngRole]
+    })
+
+    const batchComputeEnv = new batch.CfnComputeEnvironment(cluster, "batch-eks-compute-environment", {
+      type: 'MANAGED',
+      computeEnvironmentName: 'My-Eks-CE',
```

Comment:
why this name is hardcoded to `my-eks-ce`?

Reply:
addressed by adding a team prop for compute environment name